### PR TITLE
[Refactor](inverted index) add analyzer for inverted index to unify analysis process

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

The analyzer logic is currently distributed across the read, write, and no-index match phases of the inverted index. In this PR, we introduce an inverted index analyzer to unify these processes.

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index/analyzer/analyzer.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+namespace doris::segment_v2::inverted_index {
+class InvertedIndexAnalyzer {
+public:
+    static std::unique_ptr<lucene::util::Reader> create_reader(CharFilterMap& char_filter_map);
+
+    static std::unique_ptr<lucene::analysis::Analyzer> create_analyzer(
+            const InvertedIndexCtx* inverted_index_ctx);
+
+    static void get_analyse_result(std::vector<std::string>& analyse_result,
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/inverted_index/analyzer/analyzer.h:42]: std::vector<std::string>* analyse_result

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 2]
```diff
@@ -145,34 +146,15 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
...
-                get_parser_char_filter_map_from_properties(_index_meta->properties());
-        if (!char_filter_map.empty()) {
-            string_reader = std::unique_ptr<lucene::util::Reader>(CharFilterFactory::create(
-                    char_filter_map[INVERTED_INDEX_PARSER_CHAR_FILTER_TYPE],
-                    new lucene::util::SStringReader<char>(),
-                    char_filter_map[INVERTED_INDEX_PARSER_CHAR_FILTER_PATTERN],
-                    char_filter_map[INVERTED_INDEX_PARSER_CHAR_FILTER_REPLACEMENT]));
-        } else {
-            string_reader = std::make_unique<lucene::util::SStringReader<char>>();
+    Status create_char_string_reader(std::unique_ptr<lucene::util::Reader>& string_reader,
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:149]: std::unique_ptr<lucene::util::Reader>* string_reader

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 3]
```diff
@@ -207,54 +189,30 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
         return Status::OK();
     }
 
-    Status create_analyzer(std::unique_ptr<lucene::analysis::Analyzer>& analyzer) {
+    Status create_analyzer(std::unique_ptr<lucene::analysis::Analyzer>& analyzer,
```
Comment (csun5285) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:192]: std::unique_ptr<lucene::analysis::Analyzer>* analyzer

**Review by csun5285 — CHANGES_REQUESTED** [review]

**Comment by airborne12** [comment]
run buildall

**Review by csun5285 — APPROVED** [review]
LGTM

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**